servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -609,6 +609,7 @@ boolean processRole(ObjectStoreConnection con, Role originalRole, String domainN
             requestSuccess = con.updateRole(domainName, role);
         }
 
+        requestSuccess = processRoleTags(role, roleName, domainName, originalRole, con) || requestSuccess;
```

Comment:
this is actually a wrong move. we cannot go ahead and process our role tags if the role update or insert operation failed. the original order is correct.